src/BenchmarkDotNet/Engines/CacheClearingStrategies/MemoryAllocator.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Linq;
+using System.Runtime.CompilerServices;
+
+namespace BenchmarkDotNet.Engines.CacheClearingStrategies
+{
+    internal class MemoryAllocator : IMemoryAllocator
+    {
+        private const int HowManyMb = 14;
+        private const int SizeOfArray = HowManyMb * 1024 * 1024;
+        private readonly int[] memory = Enumerable.Range(1, SizeOfArray).ToArray();
+
+        public void AllocateMemory()
+        {
+            for (int i = 0; i < SizeOfArray; i++)
+            {
+                Consumer(memory[i]);
```

Comment:
Let's try to do `memory[i]++` (we will get read/write operatins instead of a single read operation).